### PR TITLE
Update API

### PR DESCRIPTION
**Pull request by andris-zalitis**

### CODE REVIEW

**Comment by rudolf-adamkovic** [comment]
In review.

File: ScrubBar/ScrubBar.swift [hunk 1]
```diff
@@ -109,33 +133,58 @@ public class ScrubBar: UIControl {
...
+    public var itemTintColor: UIColor = .gray {
+        didSet {
+            imageViews.forEach { $0.tintColor = itemTintColor }
+            selectedImageView.tintColor = selectedItemTintColor
+        }
+    }
+
+    // MARK: Selected item tint color
+
+    public var selectedItemTintColor = UIColor.darkGray {
```
Comment (rudolf-adamkovic) [inline ScrubBar/ScrubBar.swift:156]: Consider using a single style.

```swift
public var selectedItemTintColor = UIColor.darkGray { ...
```

vs

```swift
public var itemTintColor: UIColor = .gray { ...
```

File: ScrubBar/ScrubBar.swift [hunk 2]
```diff
@@ -109,33 +133,58 @@ public class ScrubBar: UIControl {
...
     public override func continueTracking(_ touch: UITouch, with event: UIEvent?) -> Bool {
-        guard let touchStartLocation = startTouchLocation else {
+        guard let startTouchLocation = startTouchLocation else {
             // don't do fatal error, because we can't ensure UIKit always calls beginTracking (though it probably does)
             return false
         }
         let location = touch.location(in: self)
-        let panDistance = abs(location.x - touchStartLocation.x)
+        let panDistance = abs(location.x - startTouchLocation.x)
         if (!isInScrubMode && panDistance >= minPanDistance) {
```
Comment (rudolf-adamkovic) [inline ScrubBar/ScrubBar.swift]: Not part of this PR but consider removing unnecessary parentheses.

File: ScrubBar/ItemLocator.swift [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+struct DefaultItemLocator: ItemLocator {
+
+    let numberOfItems: Int
+    let boundsWidth: CGFloat
+
+    init?(numberOfItems: Int, boundsWidth: CGFloat) {
+        guard numberOfItems > 0, boundsWidth > 0 else { return nil }
+
```
Comment (rudolf-adamkovic) [inline ScrubBar/ItemLocator.swift:15]: Consider removing unnecessary newline.

File: ScrubBar/ScrubBar.swift [hunk 4]
```diff
@@ -1,38 +1,62 @@
...
-    public init(images: [UIImage]) {
-        imageViews = images.map { UIImageView(image: $0) }
+    public init?(items: [ScrubBarItem]) {
+        guard items.count > 0 else { return nil }
+
+        self.items = items
+        imageViews = items.map { UIImageView(image: $0.image) }
         super.init(frame: .zero)
+
         configureImageViews()
```
Comment (rudolf-adamkovic) [inline ScrubBar/ScrubBar.swift:20]: Consider extracting `didInit()` with for the following calls:

```swift
configureImageViews()
configureStackView()
configureSelectionView()
```

Once done, consider removing unnecessary newlines in `init?`.

File: ScrubBar/ScrubBar.swift [hunk 5]
```diff
@@ -1,38 +1,62 @@
...
+    public let items: [ScrubBarItem]
+
     let imageViews: [UIImageView]
 
     private func configureImageViews() {
-        imageViews.forEach {
-            $0.contentMode = .center
-            $0.isAccessibilityElement = true
-            $0.accessibilityTraits = UIAccessibilityTraitButton
+        imageViews.enumerated().forEach { (index, element) in
```
Comment (rudolf-adamkovic) [inline ScrubBar/ScrubBar.swift]: Consider removing unnecessary parentheses.

File: ScrubBar/ScrubBar.swift [hunk 6]
```diff
@@ -1,38 +1,62 @@
...
     required public init?(coder: NSCoder) {
         return nil
     }
 
-    public var selectedSegment = 0 {
+    public var selectedIndex = 0 {
         didSet {
+            guard selectedIndex != oldValue else { return }
+
+            delegate?.scrubBar(self, didSelectItemAt: selectedIndex)
```
Comment (rudolf-adamkovic) [inline ScrubBar/ScrubBar.swift]: Consider calling `delegate` when the object is back in consistent state, i.e. end of this property observer.

File: ScrubBarTests/ScrubBarTests.swift [hunk 7]
```diff
@@ -669,3 +856,29 @@ class AnimatingMock: Animating {
 func == (lhs: AnimatingMock.AnimationConfiguration, rhs: AnimatingMock.AnimationConfiguration) -> Bool {
     return String(describing: lhs) == String(describing: rhs)
 }
+
+extension ScrubBarItem: Equatable {}
+
+extension ScrubBarItem {
+
+    static func empty() -> ScrubBarItem {
```
Comment (rudolf-adamkovic) [inline ScrubBarTests/ScrubBarTests.swift:863]: Consider replacing with an `empty` property.

**Review by rudolf-adamkovic — CHANGES_REQUESTED** [review]

File: ScrubBar/ItemLocator.swift (hunk 3 above)
Comment (andris-zalitis) [reply ScrubBar/ItemLocator.swift:15]: I think that newlines after one-line guards improve readability.

File: ScrubBarTests/ScrubBarTests.swift (hunk 7 above)
Comment (andris-zalitis) [reply ScrubBarTests/ScrubBarTests.swift:863]: ScrubBarItem has reference semantics and because of that I deliberately use a function instead of a property. That helps to illustrate that `empty()` is not idempotent.

Comment (rudolf-adamkovic) [reply ScrubBarTests/ScrubBarTests.swift:863]: @andris-zalitis 

> `ScrubBarItem` has reference semantics ...

So has `UIColor`, `NSUserDefaults` and others. It's not preferred to use a property.

Comment (andris-zalitis) [reply ScrubBarTests/ScrubBarTests.swift:863]:
```Swift
XCTAssertEqual(UIColor.red, UIColor.red)
XCTAssertEqual(UserDefaults.standard, UserDefaults.standard)
XCTAssertNotEqual(ScrubBarItem.empty(), ScrubBarItem.empty())
```

Comment (rudolf-adamkovic) [reply ScrubBarTests/ScrubBarTests.swift:863]: Then why do you need `empty()`? Why not simply `ScrubBarItem()`?

File: ScrubBar/ItemLocator.swift (hunk 3 above)
Comment (rudolf-adamkovic) [reply ScrubBar/ItemLocator.swift:15]: All right then.

**Review by rudolf-adamkovic — APPROVED** [review]
Good job!
